### PR TITLE
Update codec and crates depending

### PR DESCRIPTION
**Pull request by thiolliere**
I think we should wait on the update of scale-info https://github.com/paritytech/scale-info/pull/55

Also I haven't check if some other dependency could be updated alongside.

All in all I just update impl-codec, and then every crate that depends on it directly or indirectly and it ends up updating most crates :-/ is it expected ?

I don't need a release of all crates though I just need a release of primitive-types and impl-codec

### CODE REVIEW

File: parity-crypto/Cargo.toml
Comment (ordian) [inline parity-crypto/Cargo.toml:21]: I guess we will also need to bump `parity-crypto`

File: triehash/Cargo.toml
Comment (ordian) [inline triehash/Cargo.toml:3]: this is not required, the bump is in dev-dep

**Review by ordian — APPROVED** [review]
Thanks for taking care of this!

Could you also please update the relevant changelogs (we can publish today) and push git tags when it's merged?
The release process is documented here:
https://github.com/paritytech/parity-common/blob/master/CONTRIBUTING.md#releasing-a-new-version

**Comment by ordian** [comment]
> All in all I just update impl-codec, and then every crate that depends on it directly or indirectly and it ends up updating most crates :-/ is it expected ?

Yeah, this is really unfortunate. I'm all ears how can we decouple the dependencies. I guess we need some workaround for rust orphan rules.

File: triehash/Cargo.toml
Comment (thiolliere) [reply triehash/Cargo.toml:3]: ah my bad indeed
```suggestion
version = "0.8.4"
```

**Review by bkchr — APPROVED** [review]

File: parity-crypto/Cargo.toml
Comment (thiolliere) [reply parity-crypto/Cargo.toml:21]: yes done, thanks

**Comment by ordian** [comment]
Seems like it doesn't compile with the scale-info 0.4. We can remove this completely for now as it currently not used anywhere AFAIK.

**Review by dvdplm — APPROVED** [review]
Ty @thiolliere !

**Comment by dvdplm** [comment]
> Seems like it doesn't compile with the scale-info 0.4. We can remove this completely for now as it currently not used anywhere AFAIK.

I haven't looked yet but maybe we should release a `0.5` of scale-info.

cc @ascjones

**Comment by thiolliere** [comment]
I didn't look about other deps that could be bumped, I guess we do want to include 
https://github.com/paritytech/parity-common/pull/509/files (actually not sure about tokio)
and
https://github.com/paritytech/parity-common/pull/506/files

**Comment by ascjones** [comment]
> Seems like it doesn't compile with the scale-info 0.4. We can remove this completely for now as it currently not used anywhere AFAIK.

It will most likely be required imminently with the work I am doing to integrate `scale-info` with substrate, e.g. with`H256`

**Comment by ordian** [comment]
> I didn't look about other deps that could be bumped, I guess we do want to include
> https://github.com/paritytech/parity-common/pull/509/files (actually not sure about tokio)
> and
> https://github.com/paritytech/parity-common/pull/506/files

#506 is blocked on https://github.com/paritytech/parity-common/issues/501#issuecomment-757819969
#509 this is used in openethereum only (we should probably remove it from parity-common), they are still on old tokio

> It will most likely be required imminently with the work I am doing to integrate scale-info with substrate, e.g. withH256

please review https://github.com/paritytech/scale-info/pull/55

**Comment by thiolliere** [comment]
> push git tags when it's merged

Yes I can do the tag once merged, (but I have not rights to publish those crates)

**Comment by ordian** [comment]
> > push git tags when it's merged
> 
> Yes I can do the tag once merged, (but I have not rights to publish those crates)

you should have rights as part of core-devs member: 
* https://crates.io/teams/github:paritytech:core-devs
* https://github.com/orgs/paritytech/teams/core-devs

if you find a crate you can't publish, please report, this is fixable with 
`cargo owner --add github:paritytech:core-devs`

**Comment by thiolliere** [comment]
ah ok 👍 then I'll publish and do tags once this is merged, waiting on https://github.com/paritytech/scale-info/pull/55 for now

**Comment by ascjones** [comment]
Released https://github.com/paritytech/scale-info/pull/57

**Comment by thiolliere** [comment]
(looking at how slow is CI, I might merge and release tomorrow)

**Comment by ordian** [comment]
feel free to publish from the branch, the tests seem to pass locally
looks like travis has a long queue, but it seems to catch up: https://travis-ci.org/github/paritytech/parity-common/builds/756413352

**Pull request merged**
